h/util/group.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+"""Some shared utility functions for manipulating group data."""
+from __future__ import unicode_literals
+import re
+
+
+def split_groupid(groupid):
+    """Return the ``authority_provided_id`` and ``authority`` from a ``groupid``
+
+    For example if groupid is u'group:339ae9f33c@myauth.org' then return
+    {'authority_provided_id': u'339ae9f33c', 'authority': u'myauth.org'}'
+
+    :raises ValueError: if the given groupid isn't a valid groupid
+
+    """
+    match = re.match(r'^group:([^@]+)@(.*)$', groupid)
+    if match:
+        return {
+            'authority_provided_id': match.groups()[0],
+            'authority': match.groups()[1]
+        }
+    raise ValueError("{groupid} isn't a valid groupid".format(groupid=groupid))
```

Comment:
Where's `split_groupid` gonna end up being used? I'm wondering if it belongs as a `@staticmethod` on the `Group` class, just to keep `groupid`s themselves (the `Group.groupid` property) and this logic for working with groupids in one place?

That means that whoever wants to call `split_groupid()` needs to import (but not instantiate) the `models.Group` class. But I'm guessing the places where this is likely to be used are already going to be importing `Group`. And even if they aren't I think keeping stuff together in `Group` rather than having some groupid logic in `h.models.Group` and some groupid logic far away in `h.util.group` is a big advantage.

Reply:
@seanh I was following the established pattern in `h.util.user` in this case.